pypesto/visualize/misc.py

```diff
@@ -309,3 +285,96 @@ def apparent_composite_color_component(
         apparent_composite_color_component(fg[i], bg[i])
         for i in range(LEN_RGB)
     ]
+
+
+def process_start_indices(
+    start_indices: Union[str, int, Iterable[int]], result: Result
+):
+    """
+    Process the start_indices.
+
+    Create an array of indices if a number was provided and checks that the
+    indices do not exceed the max_index.
+
+    Parameters
+    ----------
+    start_indices:
+        list of indices or int specifying an endpoint of the sequence of
+        indices
+    result:
+        Result to determine maximum allowed length and/or clusters.
+    """
+    if start_indices is None:
+        start_indices = ALL
+    if isinstance(start_indices, str):
+        if start_indices == ALL:
+            return list(range(len(result.optimize_result)))
```

Comment:
make homogeneous whether list or array. probably np.ndarray everywhere is what you want.